### ISSUE TITLE
[bug]: [lncli] `lncli pendingsweeps` returns error `unhandled witness type for input` 

### ISSUE THREAD

**Issue by hsjoberg** [lightningnetwork/lnd#8322]
### Background

I'm running into this error when trying to run `lncli pendingsweeps` or `lncli wallet bumpclosefee`:

```
[lncli] rpc error: code = Unknown desc = unhandled witness type TaprootAnchorSweepSpend for input f490999ae4fe7b53227e57b4e8e04e854a5f6cac666da625f05da389bd1eeed5:1
```

I'm not sure what more information I can give, just ask if you need anything more.

### Your environment

* lnd version 0.17.2-beta commit=v0.17.2-beta
* Ubuntu
* bitcoind backend

Cheers
Hampus

**Comment by hsjoberg** [comment]
Just a guess, but I suppose the taproot chan anchors need to be included in the input types?

https://github.com/lightningnetwork/lnd/blob/4d8fa349ca405a42548717c17cc689e6612ff37a/lnrpc/walletrpc/walletkit_server.go#L207-L230

**Comment by hsjoberg** [comment]
It's a bit unclear to me if `txgenerator.go`'s `createSweepTx()` handles Taproot.
Presumably it does, so then just adding the taproot types to `allWitnessTypes` should work?

**Comment by ziggie1984** [comment]
Nice catch :) You are correct you need to add the new taproot types to the walletrpc enum: `WitnessType` and then add them as well to the `allWitnessTypes` mapping. A test guarantees that the `allWitnessTypes ` always reflects all types of the enum. (`TestWitnessTypeMapping`)

Do you want to do the change ?

**Comment by jimbrend** [comment]
Why are pendingsweeps related to pendingchannel close?

Why are we closing ln channels like this? 

Out of curiosity, may you ELI5?

**Comment by hsjoberg** [comment]
@jimbrend basically pendingsweeps has code to check for UTXO types, where the taproot ones are missing. So the procedure fails.

https://github.com/lightningnetwork/lnd/blob/b9a8e2f6d087c4977812034e765f7e20093b14c4/lnrpc/walletrpc/walletkit_server.go#L865-L870

**Comment by jimsbr** [comment]
Ok, thank you for confirming,

Unrelated just consolidating my thought process on other issue resolutions...

This is an ongoing issue with so many pendingchannel closures, I imagine this is also low-hanging fruit for malicious users targeting insecure lightning nodes.

I believe the remedy to be,
Documenting every instance of why we can't properly close channels including the following AFAIK:

- eclair, LND, Core Lightning, and other implementation mismatches
- One Lightning Node being offline and not able to accept the closure causing a zombie channel
- and all the nuances the above causes that requires us diving into manual intervention such as remotesweep commands AND 

chantools which no non-technical user can dare master yet with some UX troubles?

I will work on documenting a list and cross-referencing this, because this is a big issue imo. Frustrating is it not, no?

**Comment by kaloudis** [comment]
I hit the same error running `lncli wallet bumpclosefee`
